src/workers/stores/api/missionTypes.worker.ts

```diff
@@ -0,0 +1,52 @@
+import checkRequestInit from '../../../importableScripts/checkRequestInit';
+import LSSMStorage from '../../../importableScripts/LSSMStorage';
+import TypedWorker from '../../TypedWorker';
+
+import type { Mission } from 'typings/Mission';
+
+export type MissionsById = Record<Mission['id'], Mission>;
+export type MissionsArray = Mission[];
+
+interface MissionTypes {
+    missions: MissionsById;
+    missionArray: MissionsArray;
+}
+
+const scripts = { checkRequestInit, LSSMStorage } as const;
+
+export default new TypedWorker<
+    MissionTypes,
+    [locale: string, init: RequestInit],
+    Promise<MissionTypes>,
+    typeof scripts
+>(
+    'api/missionTypes.worker',
+    async (self, locale: string, init: RequestInit) => {
+        self.checkRequestInit(init);
+
+        if (self.missions)
+            return { missions: self.missions, missionArray: self.missionArray };
+
+        return fetch(`${SERVER}missions/${locale}.json`, init)
+            .then(res => res.json())
+            .then((missions: MissionsById) => {
+                self.missions = missions;
+                self.missionArray = Object.values(missions);
+                const storage = new self.LSSMStorage();
+                storage.storeMissionTypes(missions).then();
+                return { missions, missionArray: self.missionArray };
+            })
+            .catch(async () => {
```

Comment:
![grafik](https://github.com/LSS-Manager/LSSM-V.4/assets/27238105/00d93954-a900-4964-8628-3fcd313a20de)
This will now be logged (as an example I enforced to throw an error to show how it would look like) when an Error occurs in a Worker. The log will happen on the context that requested the processes.

Browsers _may_ additionally show the error thrown in the worker within the context that initially created the worker which _may_ lead to duplicated error reports:
![grafik](https://github.com/LSS-Manager/LSSM-V.4/assets/27238105/8bdb9b3d-c287-42a4-ad00-b40305ea1799)

This is online with commit ae7912a.